numpy/core/src/umath/scalarmath.c.src

```diff
@@ -569,6 +570,91 @@ static void
 
 /*** END OF BASIC CODE **/
 
+/**
+ * Find the descriptor for a builtin or Python numerical scalar.
+ * This function should only be used for scalar math fast paths.
+ *
+ * @param obj The object for which to find the descriptor
+ * @param descr The descriptor that was found.
+ * @returns 1 if the other object is a Python type and thus must not
+ *          handle the operation, otherwise 0. -1 _without_ an error set
+ *          if no descriptor was found.
+ */
+static int
+descr_from_basic_scalar(PyObject *obj, PyArray_Descr **descr)
+{
+    /* TODO: We could try giving defined scalars a chance... */
+    int type_num = PyArray_TypeNumFromNumPyScalarType(
+            (PyObject *)Py_TYPE(obj), 0);
+    if (type_num != NPY_NOTYPE) {
+        *descr = PyArray_DescrFromType(type_num);
+        return 0;
+    }
+    else if (PyFloat_CheckExact(obj)) {
```

Comment:
We could add an error check here as well?

Reply:
Sorry, where do you mean? The type check and `DescrFromType` cannot fail here (the latter is a bit obscure sometimes, I admit, but still cannot fail since it returns a singleton).

Reply:
```suggestion
    else if (PyFloat_CheckExact(obj)) {
        if (PyFloat_AsDouble(obj) < 0 && PyErr_Occurred()) {
            PyErr_Clear();
            return -1;
        }
```
So I mean like this, to make sure the object can be interpreted safely as a `float`

Reply:
`PyFloat_AsDouble` cannot possibly fail after the `PyFloat_CheckExact`, at least in cpython. So I don't think we have to add it.

Reply:
Ohh thanks, I did not know that.